File: Tsk3/C/Subtask 3/Subtask3.c
#include <stdio.h>

int main() {
    int n;
    printf("Enter a value: ");
    scanf("%d", &n);

    if (n % 2 == 0) {
        n--;
    }

    int k = (n + 1) / 2 - 1;

    for (int i=1; i <= n; i += 2) {
        for (int j = 0; j < k; ++j) {
            printf(" ");
        }
        for (int j = 0; j < i; ++j) {
            printf("*");
        }
        printf("\n");
        k--;
    }

    k = 1;
    for (int i = n-2; i >0; i -= 2) {
        for (int j= 0; j<k; ++j) {
            printf(" ");
        }
        for (int j = 0; j< i; ++j) {
            printf("*");
        }
        printf("\n");
        k++;
    }

    return 0;
}

File: Tsk3/Java/Subtask 3/Subtask3.java
import java.util.Scanner;

public class Subtask3 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a val: ");
        int n = scanner.nextInt();
        if (n % 2 == 0) {
            n -= 1;
        }
        int k = (n + 1) - 2;
        for (int i = 1; i <= n; i += 2) {
            k -= 1;
            printPatternLine(k, i);
        }
        for (int i = n - 2; i > 0; i -= 2) {
            k += 1;
            printPatternLine(k, i);
        }
        scanner.close();
    }

    private static void printPatternLine(int spaces, int stars) {
        for (int j = 0; j < spaces; j++) {
            System.out.print(' ');
        }
        for (int j = 0; j < stars; j++) {
            System.out.print('*');
        }
        System.out.println();
    }
}
